File: 128_longest_consecutive_sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::unordered_set<int> myset;
        int maxlen = 0;
        for(int i =0; i<nums.size();i++)
            myset.insert(nums[i]);
        for(int i =0; i<nums.size();i++)
        {
            if (myset.find(nums[i]-1) == myset.end() && myset.find(nums[i]) != myset.end() && myset.find(nums[i]+1) != myset.end()){
                int count=1, start = nums[i];
                for (start; count < nums.size() && myset.find(start+1) != myset.end(); start++)
                    count++;
                maxlen = std::max(maxlen, count);
            }
        }
        if (myset.size()>0 && maxlen==0)
            return 1;
        else
            return maxlen;
    }
};

File: 128_longest_consecutive_sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        setn = set(nums)
        maxlen = 0
        for i in nums:
            if i-1 not in setn and i in setn and i+1 in setn:
                count = 1
                start = i
                while count < len(nums) and start+1 in setn:
                    start+=1
                    count+=1
                maxlen = max(count,maxlen)
        if maxlen==0 and len(setn)>0:
            return 1
        return maxlen